module/VuFind/src/VuFind/Marc/Serialization/MarcXml.php

```diff
@@ -48,29 +48,52 @@ class MarcXml implements SerializationInterface
     public static function canParse(string $marc): bool
     {
         // A pretty naïve check, but it's enough to tell the different formats apart
-        return strncmp($marc, '<', 1) === 0;
+        return strncmp(trim($marc), '<', 1) === 0;
+    }
+
+    /**
+     * Check if the serialization class can parse the given MARC collection string
+     *
+     * @param string $marc MARC
+     *
+     * @return bool
+     */
+    public static function canParseCollection(string $marc): bool
+    {
+        // A pretty naïve check, but it's enough to tell the different formats apart
+        return strncmp(trim($marc), '<', 1) === 0;
+    }
+
+    /**
+     * Parse MARC collection from a string into an array
+     *
+     * @param string $collection MARC record collection in the format supported by
+     * the serialization class
+     *
+     * @throws \Exception
+     * @return array
+     */
+    public static function collectionFromString(string $collection): array
+    {
+        $xml = static::loadXML(trim($collection));
+        $results = [];
+        foreach ($xml->record as $record) {
+            $results[] = $record->saveXML();
```

Comment:
Any strong reason to use `saveXML` here instead of `asXML`? I prefer `asXML` (though the two are synonymous). It looks like both forms are currently used in the VuFind codebase, but there are more `asXML`'s than `saveXML`'s. Perhaps I should open a PR to normalize them.

Reply:
Not really (apart from the fact that DOMDocument only has `saveXML`, but there's no need for unity here). I'll change it.